habitat-lab/habitat/gui/text_drawer.py

```diff
@@ -0,0 +1,94 @@
+import os
+import string
+
+import magnum as mn
+from magnum import shaders, text
+
+DEFAULT_FONT_PATH = "./fonts/ProggyClean.ttf"
+
+# the maximum number of chars displayable in the app window
+# using the magnum text module
+MAX_DISPLAY_TEXT_CHARS = 1024
+
+# how much to displace window text relative to the center of the
+# app window (e.g if you want the display text in the top left of
+# the app window, you will displace the text
+# window width * -TEXT_DELTA_FROM_CENTER in the x axis and
+# window height * TEXT_DELTA_FROM_CENTER in the y axis, as the text
+# position defaults to the middle of the app window)
+TEXT_DELTA_FROM_CENTER = 0.49
+
+# font size of the magnum in-window display text
+DISPLAY_FONT_SIZE = 16.0
+
+
+class TextDrawer:
+    def __init__(
+        self,
+        framebuffer_size: mn.Vector2i,
+        relative_path_to_font: str = DEFAULT_FONT_PATH,
+        max_display_text_chars: int = MAX_DISPLAY_TEXT_CHARS,
+        text_delta_from_center: float = TEXT_DELTA_FROM_CENTER,
+        display_font_size: float = DISPLAY_FONT_SIZE,
+    ) -> None:
+        self._framebuffer_size = framebuffer_size
+
+        # Load a TrueTypeFont plugin and open the font file
+        self._display_font = text.FontManager().load_and_instantiate(
```

Comment:
@rpartsey can you post a link to where you borrowed this code from?

@mosra I wanted you to have a look at this Magnum text-drawing code.

Reply:
All text drawing code borrowed / copy-pasted from [habitat-sim/examples/viewer.py](https://github.com/facebookresearch/habitat-sim/blob/100dc8fb140507a3b63565f69e5da2bcbf91f426/examples/viewer.py#L112).